applications/views.py

```diff
@@ -1,15 +1,31 @@
 from django.contrib.auth.decorators import login_required
+from django.http import Http404
 from django.shortcuts import get_object_or_404, redirect
 from django.template.response import TemplateResponse
 
+from jobserver.authorization import has_permission
+
 from .form_specs import form_specs
 from .models import Application
 from .wizard import Wizard
 
 
+def validate_application_access(user, application):
+    if has_permission(user, "application_manage"):
+        return
+
+    if application.created_by == user:
+        return
+
+    raise Http404
```

Comment:
This might be the right solution for now, but looking at `application.created_by` is a new way of managing per-object permissions, and is different to the existing `RolesField` stuff.

Reply:
There's precedent with how we decide who can cancel a Job too, it spawned the conversation in #795